patches/0003-Add-used-attribute-and-custom-memory-address-for-gco.patch

```diff
@@ -0,0 +1,31 @@
+From 8dfcb8372fb9007459ce385e8110d058fd4db418 Mon Sep 17 00:00:00 2001
+From: Eduard Mihailescu <mihailescu.eduard@gmail.com>
+Date: Thu, 6 Apr 2023 19:08:35 +0300
+Subject: [PATCH 4/4] Add 'used' attribute and custom memory address for
+ gcov_output_bufferr
+
+---
+ code/gcov_public.c | 7 +++++--
+ 1 file changed, 5 insertions(+), 2 deletions(-)
+
+diff --git a/code/gcov_public.c b/code/gcov_public.c
+index 94aaff4..50e7e66 100644
+--- a/code/gcov_public.c
++++ b/code/gcov_public.c
+@@ -74,8 +74,11 @@ typedef unsigned int u32;
+ /* You need to set the output buffer pointer to your memory block */
+ /* Size used will depend on size and complexity of source code
+  * that you have compiled for coverage. */
+-static unsigned char *gcov_output_buffer = (unsigned char *)(0x42000000);
+-static gcov_unsigned_t gcov_output_index;
++static unsigned char __attribute__((used)) *gcov_output_buffer = (unsigned char *) 0x42000000;
++#ifdef BUFFER_ADDRESS
++static unsigned char __attribute__((used)) *gcov_output_buffer = (unsigned char *) BUFFER_ADDRESS;
```

Comment:
Here you do more things than what you describe in the commit message, ie introduce a new config option to adjust the base address of the buffer used in memory output. A strict reviewer would had asked you to provide these changes in separate commits. For the scope of this PR this is fine, but please update the commit message.